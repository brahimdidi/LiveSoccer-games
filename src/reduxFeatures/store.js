import { configureStore } from '@reduxjs/toolkit';
import allGamesReducer from './allGamesSlice';

const store = configureStore({
  reducer: {
    allGames: allGamesReducer,
  },
});

export default store;
