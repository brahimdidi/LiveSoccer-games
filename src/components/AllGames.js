import { useDispatch, useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
import { NavLink } from 'react-router-dom';
import { MdReadMore } from 'react-icons/md';
import { fetchAllGames, selectAllGames } from '../reduxFeatures/allGamesSlice';
import TodayGames from './TodayGames';

function AllGames() {
  const allGamesList = useSelector(selectAllGames);
  // styles for the navbar elements
  const [todayGames, setTodayGames] = useState(false);
  const [allGames, setAllGames] = useState(true);

  const dispatch = useDispatch();

  useEffect(() => {
    if (!allGamesList.length)dispatch(fetchAllGames());
  }, [dispatch]);
  const handleTodayGames = () => {
    setTodayGames(true);
    setAllGames(false);
  };
  const handleAllGames = () => {
    setTodayGames(false);
    setAllGames(true);
  };
  return (
    <div className="allGames-app">
      <nav className="allGames-nav">
        <button
          className={`btn ${allGames ? 'btn-active' : ''}`}
          type="button"
          onClick={handleAllGames}
        >
          All Games
        </button>
        <button
          className={`btn ${todayGames ? 'btn-active' : ''}`}
          type="button"
          onClick={handleTodayGames}
        >
          TodayGames
        </button>
      </nav>
      <ul className="games-container">
        {!todayGames && allGamesList.map((game) => (
          <li key={game.id}>
            <div className="competition">
              <h1>
                {game.competition}
              </h1>
              <NavLink className="AllGames-to-Competition" to="/Competition" state={{ league: game.competition }}>
                <MdReadMore />
              </NavLink>
            </div>

            <h2 className="date">{game.date}</h2>
            <div className="title-time">
              <h3>{game.title}</h3>
              <span>{game.time}</span>
            </div>

          </li>

        ))}
        {todayGames && <TodayGames list={allGamesList} />}
        {!allGamesList.length && <h1 className="text-danger">Check your internet Again</h1>}
      </ul>
    </div>
  );
}

export default AllGames;
