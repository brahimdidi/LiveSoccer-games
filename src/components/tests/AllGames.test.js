import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../reduxFeatures/store';
import AllGames from '../AllGames';

describe('AllGames Test', () => {
  it('renders correctly', () => {
    const AllGamesTest = renderer.create(
      <Provider store={store}>
        <Router>
          <AllGames />
        </Router>
      </Provider>,
    ).toJSON();
    expect(AllGamesTest).toMatchSnapshot();
  });
});
