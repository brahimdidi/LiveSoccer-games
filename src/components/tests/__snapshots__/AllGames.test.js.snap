// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllGames Test renders correctly 1`] = `
<div
  className="allGames-app"
>
  <nav
    className="allGames-nav"
  >
    <button
      className="btn btn-active"
      onClick={[Function]}
      type="button"
    >
      All Games
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      TodayGames
    </button>
  </nav>
  <ul
    className="games-container"
  >
    <h1
      className="text-danger"
    >
      Check your internet Again
    </h1>
  </ul>
</div>
`;
